File: lib/src/models/product_review.dart
class ProductReview {
  final int reviewId;
  final int productId;
  final double rating;
  final String comment;
  final int userId;
  final DateTime createdAt;

  ProductReview({
    required this.reviewId,
    required this.productId,
    required this.rating,
    required this.comment,
    required this.userId,
    required this.createdAt
  });

  factory ProductReview.fromJson(Map<String, dynamic> json) {
    return ProductReview(
      reviewId: json['reviewId'],
      productId: json['productId'],
      rating: json['rating'],
      comment: json['comment'],
      userId: json['userId'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}

File: src/main/java/com/example/sneakerx/entities/ProductReview.java
package com.example.sneakerx.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "product_reviews")
public class ProductReview {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer reviewId;

    private Double rating;
    private String comment;
    private Integer userId;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing the Cart back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private Product product;
}
